lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -217,14 +246,26 @@ export class AuthCache extends BrowserStorage {// Singleton
      * @param state
      */
     public static generateAcquireTokenAccountKey(accountId: any, state: string): string {
-        return `${TemporaryCacheKeys.ACQUIRE_TOKEN_ACCOUNT}${Constants.resourceDelimiter}${accountId}${Constants.resourceDelimiter}${state}`;
+        const stateId = RequestUtils.parseLibraryState(state).id;
+        return `${TemporaryCacheKeys.ACQUIRE_TOKEN_ACCOUNT}${Constants.resourceDelimiter}${accountId}${Constants.resourceDelimiter}${stateId}`;
     }
 
     /**
      * Create authorityKey to cache authority
      * @param state
      */
     public static generateAuthorityKey(state: string): string {
```

Comment:
what's the benefit of having both `generateAuthorityKey` and `generateTemporaryCacheKey`?

Reply:
It was already there and decided not to remove it. Just a convenience method.